docs/reference/components/transforms/remap.cue

```diff
@@ -0,0 +1,124 @@
+package metadata
+
+components: transforms: remap: {
+  title: "Remap"
+  description: "Maps and parses log fields using a light-weight, fast, and robust syntax."
+
+  _features: {
+  }
+
+  classes: {
+    commonly_used: true
+    function: "map"
+  }
+
+  statuses: {
+    development: "beta"
+  }
+
+  support: {
+    platforms: {
+      "aarch64-unknown-linux-gnu": true
+      "aarch64-unknown-linux-musl": true
+      "x86_64-pc-windows-msv": true
+      "x86_64-unknown-linux-gnu": true
+      "x86_64-unknown-linux-musl": true
+    }
+
+    requirements: []
+    warnings: []
+  }
+
+  configuration: {
+    source: {
+      common: true
+      description: "The remap source/instruction set to execute for each event"
+      required: true
+      type: string: {
+        examples: [
+          """
+          .type = "foo",
+          .new_field = .old_field * 2
+          del(.old_field)
+          """
+        ]
+      }
+    }
+  }
+
+  examples: log: [
+    {
+      title: "Add, Rename, & Remove Fields"
+      configuration: {
+        source: #"""
+                .new_field = "new value"
+                .new_name = .old_name
+                del(.old_name)
+                """#
+      }
+      input: {
+        old_name: "old value"
+      }
+      output: {
+        new_field: "new value"
+        new_name: "old value"
+      }
+    },
+    {
+      title: "Parse JSON"
+      configuration: {
+        source: #"""
+                message = del(.message)
```

Comment:
The `del` (and `only_fields`, and `merge`) function does not have a return value and thus can't be assigned to a path (or variable, if we supported those).

We're thinking of changing that though (https://github.com/timberio/vector/pull/3927#discussion_r489313588).